### PR TITLE
Return 401 instead of 500 if JWT expired.

### PR DESCRIPTION
**Pull request by chenrujun**
- Return `401` instead of `500` if JWT expired.

### CODE REVIEW

**Comment by chenrujun** [comment]
Hi, @saragluna , please help to review this PR. 🙏

File: sdk/spring/azure-spring-boot/src/test/java/com/microsoft/azure/spring/autoconfigure/aad/AADAppRoleAuthenticationFilterTest.java [hunk 1]
```diff
@@ -105,8 +105,8 @@ public void testDoFilterGoodCase()
         assertNull("Authentication has not been cleaned up!", SecurityContextHolder.getContext().getAuthentication());
     }
 
-    @Test(expected = ServletException.class)
-    public void testDoFilterShouldRethrowJWTException()
+    @Test
+    public void testDoFilterShouldNotRethrowJWTException()
```
Comment (saragluna) [inline sdk/spring/azure-spring-boot/src/test/java/com/microsoft/azure/spring/autoconfigure/aad/AADAppRoleAuthenticationFilterTest.java:109]: Do we need to add some verification logic here?

**Review by saragluna — COMMENTED** [review]
Should we add some verification logic to the Bad JWT case?

File: sdk/spring/azure-spring-boot/src/test/java/com/microsoft/azure/spring/autoconfigure/aad/AADAppRoleAuthenticationFilterTest.java (hunk 1 above)
Comment (chenrujun) [reply sdk/spring/azure-spring-boot/src/test/java/com/microsoft/azure/spring/autoconfigure/aad/AADAppRoleAuthenticationFilterTest.java]: Updated.

**Review by saragluna — APPROVED** [review]
LGTM.

**Pull request merged**